### PR TITLE
add tests and documentation

### PR DESCRIPTION
**Pull request by tomascorrea**
Issue 1740 add test quick tutorial 20 https://github.com/Pylons/pyramid/issues/1740
Add tests and documentation to quick tutorial.

### CODE REVIEW

File: docs/quick_tutorial/authentication/setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 requires = [
     'bcrypt',
-    'pyramid',
+    'pyramid[testing]',
```
Comment (mmerickel) [inline docs/quick_tutorial/authentication/setup.py]: same comment as https://github.com/Pylons/pyramid/pull/3376#discussion_r223190034

Comment (stevepiercy) [reply docs/quick_tutorial/authentication/setup.py]: I standardized setup.py throughout the Quick Tutorial, and used the GitHub UI to clean it up. Resolving conversation.

File: docs/quick_tutorial/authentication.rst [hunk 2]
```diff
@@ -101,6 +101,31 @@ Steps
...
+
+       $VENV/bin/pip install -e ".[testing]"
+
+#. Run your Pyramid application tests with:
+
+   .. code-block:: bash
+
+       $VENV/bin/pytest tutorial/tests.py
+
+
```
Comment (stevepiercy) [inline docs/quick_tutorial/authentication.rst]: Please rebase your changes on (or manually compare to) `master`, per the updates I recently made throughout the Quick Tutorial's setup.py's and narrative documentation, then push your changes to this branch.

File: docs/quick_tutorial/authentication.rst [hunk 3]
```diff
@@ -101,6 +101,31 @@ Steps
...
+    :language: python
+    :emphasize-lines: 11
+    :linenos:
+
+#. Install testing dependency:
+
+   .. code-block:: bash
+
+       $VENV/bin/pip install -e ".[testing]"
+
```
Comment (stevepiercy) [inline docs/quick_tutorial/authentication.rst]: Please remove lines 109-115. I think my recent setup.py changes will handle this.

Also please check that my narrative documentation changes do not already handle line 120. Else change line 120 to `$VENV/bin/pip install -e ".[test]"`.

Comment (stevepiercy) [reply docs/quick_tutorial/authentication.rst]: The tests look good. Thank you for your work on these!

Also please sign CONTRIBUTORS.txt, if you have not done so.

File: docs/quick_tutorial/authentication.rst [hunk 4]
```diff
@@ -35,7 +35,7 @@ Steps
 
        cd ..; cp -r view_classes authentication; cd authentication
 
-#. Add ``bcrypt`` as a dependency in ``authentication/setup.py``:
+#. Add ``bcrypt`` and ``pyramid[pyramid]`` as a dependency in ``authentication/setup.py``:
```
Comment (mmerickel) [inline docs/quick_tutorial/authentication.rst]: what is `pyramid[pyramid]`? This is definitely not correct but I can't tell what it's even supposed to be.

**Comment by mmerickel** [comment]
I'm going to close this soon due to inactivity.

**Pull request closed without merging**